realis.go

```diff
@@ -62,13 +63,146 @@ type realisClient struct {
 	readonlyClient *aurora.ReadOnlySchedulerClient
 }
 
+type option func(*RealisConfig)
+
+//Config sets for options in RealisConfig.
+func BasicAuth(username, password string) option {
+
+	return func(config *RealisConfig) {
+		config.username = username
+		config.password = password
+	}
+}
+
+func SchedulerUrl(url string) option {
+	return func(config *RealisConfig) {
+		config.url = url
+	}
+}
+
+func TimeoutMS(timeout int) option {
+	return func(config *RealisConfig) {
+		config.timeoutms = timeout
+	}
+}
+
+func ZKCluster(cluster *Cluster) option {
+	return func(config *RealisConfig) {
+		config.cluster = cluster
+	}
+}
+
+func ZKUrl(url string) option {
+	return func(config *RealisConfig) {
+		config.cluster = GetDefaultClusterFromZKUrl(url)
+	}
+}
+
+func Retries(backoff *Backoff) option {
+	return func(config *RealisConfig) {
+		config.backoff = backoff
+	}
+}
+
+func ThriftJSON() option {
+	return func(config *RealisConfig) {
+		config.jsonTransport = true
+	}
+}
+
+func ThriftBinary() option {
+	return func(config *RealisConfig) {
+		config.binTransport = true
+	}
+}
+
+func BackOff(b *Backoff) option {
+	return func(config *RealisConfig) {
+		config.backoff = b
+	}
+}
+
+func NewRealisClient(options ...option) (Realis, error) {
+	config := &RealisConfig{}
+	fmt.Println(" options length: ", options)
+	for _, opt := range options {
+		opt(config)
+	}
+	//Default timeout
+	if config.timeoutms == 0 {
+		config.timeoutms = 10000
+	}
+	//Set default Transport to JSON if needed.
+	if !config.jsonTransport && !config.binTransport {
+		config.jsonTransport = true
```

Comment:
If we we move this an line 133 before we apply options (line 128), could we do away with the if statements to simply code a bit?